diff --git a/tutorial/tutorial_implicit.input b/tutorial/tutorial_implicit.input
index 13a5c94..1780803 100644
--- a/tutorial/tutorial_implicit.input
+++ b/tutorial/tutorial_implicit.input
@@ -3,5 +3,5 @@ TEnd = 500000 # duration of the simulation [s]
 DtInitial = 10 # initial time step size [s]
 
 [Grid]
-UpperRight = 300 60 # x-/y-coordinates of the upper-right corner of the grid [m]
-Cells = 100 1 # x-/y-resolution of the grid
+UpperRight = 400 500 # x-/y-coordinates of the upper-right corner of the grid [m]
+Cells = 20 25 # x-/y-resolution of the grid
