diff --git a/dune/istl/paamg/amg.hh b/dune/istl/paamg/amg.hh
index 56b1d8d..d672756 100644
--- a/dune/istl/paamg/amg.hh
+++ b/dune/istl/paamg/amg.hh
@@ -526,13 +526,13 @@ namespace Dune
               // We are still participating on this level
               solver_.reset(new BiCGSTABSolver<X>(const_cast<M&>(matrices_->matrices().coarsest().getRedistributed()),
                                                   *scalarProduct_,
-                                                  *coarseSmoother_, 1E-2, 1000, 0));
+                                                  *coarseSmoother_, 1E-12, 1000, 0));
             else
               solver_.reset();
           }else
             solver_.reset(new BiCGSTABSolver<X>(const_cast<M&>(*matrices_->matrices().coarsest()),
                                                 *scalarProduct_,
-                                                *coarseSmoother_, 1E-2, 1000, 0));
+                                                *coarseSmoother_, 1E-12, 1000, 0));
         }
       }
 
