diff --git a/opm-grid-prereqs.cmake b/opm-grid-prereqs.cmake
index 2ad0d86..032ac19 100644
--- a/opm-grid-prereqs.cmake
+++ b/opm-grid-prereqs.cmake
@@ -34,4 +34,5 @@ set (opm-grid_DEPS
   "ecl REQUIRED"
   )
 
+include(OpmPackage)
 find_package_deps(opm-grid)
diff --git a/opm/grid/common/p2pcommunicator.hh b/opm/grid/common/p2pcommunicator.hh
index 1b4b327..55a2b2f 100644
--- a/opm/grid/common/p2pcommunicator.hh
+++ b/opm/grid/common/p2pcommunicator.hh
@@ -23,6 +23,7 @@
 #include <vector>
 #include <set>
 #include <map>
+#include <cassert>
 
 #include <dune/common/version.hh>
 
