diff --git a/cmake/Modules/Findopm-parser.cmake b/cmake/Modules/Findopm-parser.cmake
index cb08ade..612f669 100644
--- a/cmake/Modules/Findopm-parser.cmake
+++ b/cmake/Modules/Findopm-parser.cmake
@@ -38,8 +38,8 @@ string(REGEX REPLACE "${PROJECT_SOURCE_DIR}/?(.*)" "\\1"  BUILD_DIR_SUFFIX "${PR
 # or in relative directories to this one
 if (OPM_PARSER_ROOT)
   set (_no_default_path "NO_DEFAULT_PATH")
-  set (_opm_parser_source "")
-  set (_opm_parser_build "")
+  set (_opm_parser_source "${OPM_PARSER_ROOT}")
+  set (_opm_parser_build "${OPM_PARSER_ROOT}/${BUILD_DIR_SUFFIX}")
 else ()
   set (_no_default_path "")
   set (_opm_parser_source
