diff --git a/CMakeLists_files.cmake b/CMakeLists_files.cmake
index fbbdaf05..34cdc425 100644
--- a/CMakeLists_files.cmake
+++ b/CMakeLists_files.cmake
@@ -36,7 +36,7 @@ list (APPEND MAIN_SOURCE_FILES
         opm/core/flowdiagnostics/TofReorder.cpp
         opm/core/linalg/LinearSolverFactory.cpp
         opm/core/linalg/LinearSolverInterface.cpp
-        opm/core/linalg/LinearSolverIstl.cpp
+#        opm/core/linalg/LinearSolverIstl.cpp
         opm/core/linalg/LinearSolverPetsc.cpp
         opm/core/linalg/LinearSolverUmfpack.cpp
         opm/core/linalg/call_umfpack.c
@@ -133,12 +133,12 @@ list (APPEND TEST_SOURCE_FILES
 	tests/test_nonuniformtablelinear.cpp
 	tests/test_parallelistlinformation.cpp
 	tests/test_sparsevector.cpp
-       tests/test_velocityinterpolation.cpp
+    tests/test_velocityinterpolation.cpp
 	tests/test_uniformtablelinear.cpp
 	tests/test_wells.cpp
 	tests/test_wachspresscoord.cpp
-	tests/test_linearsolver.cpp
-	tests/test_parallel_linearsolver.cpp
+#	tests/test_linearsolver.cpp
+#	tests/test_parallel_linearsolver.cpp
 	tests/test_param.cpp
 	tests/test_satfunc.cpp
 	tests/test_shadow.cpp
@@ -153,7 +153,7 @@ list (APPEND TEST_SOURCE_FILES
 	tests/test_anisotropiceikonal.cpp
 	tests/test_stoppedwells.cpp
 	tests/test_relpermdiagnostics.cpp
-        tests/test_norne_pvt.cpp
+    tests/test_norne_pvt.cpp
   )
 
 # originally generated with the command:
@@ -194,8 +194,8 @@ list (APPEND TEST_DATA_FILES
 list (APPEND EXAMPLE_SOURCE_FILES
 	examples/compute_eikonal_from_files.cpp
 	examples/compute_initial_state.cpp
-	examples/compute_tof.cpp
-	examples/compute_tof_from_files.cpp
+#	examples/compute_tof.cpp
+#	examples/compute_tof_from_files.cpp
 	examples/diagnose_relperm.cpp
 	tutorials/tutorial1.cpp
 	tutorials/tutorial2.cpp
