diff --git a/CMakeLists.txt b/CMakeLists.txt
index d81c5dc2..964c19de 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -91,14 +91,7 @@ else ()
     add_subdirectory(external/cjson)
 endif()
 
-# if building shared libraries, then don't include Boost in them
-if (BUILD_SHARED_LIBS)
-    set(Boost_USE_STATIC_LIBS OFF)
-    set(Boost_USE_STATIC_RUNTIME OFF)
-else ()
-# if using dynamic boost, the header file must generate a main() function
-   set(Boost_USE_STATIC_LIBS ON)
-endif ()
+add_definitions(-DBOOST_TEST_DYN_LINK)
 
 find_package(Boost 1.44.0
              COMPONENTS filesystem
