diff --git a/CMakeLists.txt b/CMakeLists.txt
index a53c6634..2a6b5147 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -99,24 +99,8 @@ SET( CMAKE_C_FLAGS_DEBUG   "-O0 -DDEBUG -ggdb3         ${CMAKE_C_FLAGS} ${CMAKE_
 SET( CMAKE_C_FLAGS_RELEASE "-O3 -DNDEBUG -mtune=native ${CMAKE_C_FLAGS} ${CMAKE_C_FLAGS_RELEASE}")
 
 
-if (BOOST_ROOT AND NOT DEFINED Boost_NO_SYSTEM_PATHS)
-   set (Boost_NO_SYSTEM_PATHS TRUE)
-endif (BOOST_ROOT AND NOT DEFINED Boost_NO_SYSTEM_PATHS)
-
-# if we are building shared libraries ourselves, then don't include Boost in them
-if (BUILD_SHARED_LIBS)
-   set(Boost_USE_STATIC_LIBS    OFF)
-elseif (DEFINED BUILD_SHARED_LIBS)
-   set(Boost_USE_STATIC_LIBS    ON)
-endif ()
-set(Boost_USE_MULTITHREADED      ON)
-set(Boost_USE_STATIC_RUNTIME    OFF)
-
-
-add_definitions(-DOPM_PARSER_DECK_API=1)
-# Requires BOOST filesystem version 3, thus 1.44 is necessary.
-add_definitions(-DBOOST_FILESYSTEM_VERSION=3)
-find_package(Boost 1.44.0 COMPONENTS filesystem date_time system unit_test_framework regex REQUIRED)
+# get the prerequisite Boost libraries
+find_package(Boost 1.44.0 COMPONENTS filesystem date_time system unit_test_framework regex ${OPM_PARSER_QUIET})
 include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
 
 include_directories(BEFORE ${PROJECT_SOURCE_DIR})
@@ -177,9 +161,9 @@ include(OpmProject)
 include(ConfigVars)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(opm-parser_NAME opm-parser)
-set(opm-parser_LIBRARIES opmparser opmjson)
-if(NOT BUILD_SHARED_LIBS)
-  list(APPEND opm-parser_LIBRARIES cjson)
-endif()
+#set(opm-parser_LIBRARIES opmparser opmjson)
+#if(NOT BUILD_SHARED_LIBS)
+#  list(APPEND opm-parser_LIBRARIES cjson)
+#endif()
 set(opm-parser_TARGET opmparser)
 opm_cmake_config(opm-parser)
