Index: dune/pdelab/backend/istlvectorbackend.hh
===================================================================
--- dune/pdelab/backend/istlvectorbackend.hh	(revision 2292)
+++ dune/pdelab/backend/istlvectorbackend.hh	(working copy)
@@ -30,6 +34,8 @@
       typedef typename ContainerType::size_type size_type;
       typedef ISTLVectorBackend<BLOCKSIZE> Backend;
 
+      ISTLBlockVectorContainer ()
+      {}
       ISTLBlockVectorContainer (const T& t_) : container(t_.globalSize()/BLOCKSIZE)
       {}
       ISTLBlockVectorContainer (const T& t_, const E& e) : container(t_.globalSize()/BLOCKSIZE)
@@ -89,6 +95,11 @@
         return container.two_norm();
       }
 
+      typename Dune::template FieldTraits<E>::real_type two_norm2() const
+      {
+        return container.two_norm2();
+      }
+
       typename Dune::template FieldTraits<E>::real_type one_norm() const
       {
         return container.one_norm();
@@ -163,6 +174,16 @@
         return container.size()*BLOCKSIZE;
       }
 
+      size_t size() const
+      {
+        return container.size();
+      }
+
+      void resize(size_t n) 
+      {
+        container.resize(n);
+      }
+
       template<typename X>
       void std_copy_to (std::vector<X>& x) const
       {
Index: dune/pdelab/backend/novlpistlsolverbackend.hh
===================================================================
--- dune/pdelab/backend/novlpistlsolverbackend.hh	(revision 2292)
+++ dune/pdelab/backend/novlpistlsolverbackend.hh	(working copy)
@@ -1251,7 +1251,7 @@
                             int verbose_=1, bool reuse_=false,
                             bool usesuperlu_=true)
         : gfs(gfs_), phelper(gfs,verbose_), maxiter(maxiter_),
-          params(15,2000,1.2,1.6,Dune::Amg::atOnceAccu),
+          params(15,2000,1.2,1.6,Dune::Amg::noAccu/*Dune::Amg::atOnceAccu*/),
           verbose(verbose_), reuse(reuse_), firstapply(true),
           usesuperlu(usesuperlu_)
       {
@@ -1314,10 +1314,10 @@
         \param[in] r right hand side
         \param[in] reduction to be achieved
       */
-      void apply(M& A, V& z, V& r, typename V::ElementType reduction)
+      void apply(MatrixType& A, V& z, V& r, typename V::ElementType reduction)
       {
         Timer watch;
-        MatrixType& mat=A.base();
+        MatrixType& mat=A;
         typedef Dune::Amg::CoarsenCriterion<Dune::Amg::SymmetricCriterion<MatrixType,
           Dune::Amg::FirstDiagonal> > Criterion;
 #if HAVE_MPI
Index: dune/pdelab/backend/ovlpistlsolverbackend.hh
===================================================================
--- dune/pdelab/backend/ovlpistlsolverbackend.hh	(revision 2292)
+++ dune/pdelab/backend/ovlpistlsolverbackend.hh	(working copy)
@@ -803,7 +803,7 @@
       ISTLBackend_AMG(const GFS& gfs_, unsigned maxiter_=5000, 
                       int verbose_=1, bool reuse_=false,
                       bool usesuperlu_=true)
-        : gfs(gfs_), phelper(gfs,verbose_), maxiter(maxiter_), params(15,2000),
+        : gfs(gfs_), phelper(gfs,verbose_), maxiter(maxiter_), params(15,2000,1.2,1.6,Dune::Amg::atOnceAccu),
           verbose(verbose_), reuse(reuse_), firstapply(true),
           usesuperlu(usesuperlu_)
       {
@@ -864,11 +864,12 @@
         \param[in] r right hand side
         \param[in] reduction to be achieved
       */
-      void apply(M& A, V& z, V& r, typename V::ElementType reduction)
+      template <class Vector>
+      void apply(MatrixType& A, Vector& z, Vector& r, typename V::ElementType reduction)
       {
         Timer watch;
         Comm oocc(gfs.gridView().comm());
-        MatrixType& mat=A.base();
+        MatrixType& mat=A;
         typedef Dune::Amg::CoarsenCriterion<Dune::Amg::SymmetricCriterion<MatrixType,
           Dune::Amg::FirstDiagonal> > Criterion;
 #if HAVE_MPI
@@ -900,8 +901,7 @@
         Solver<VectorType> solver(oop,sp,*amg,reduction,maxiter,verb);
         Dune::InverseOperatorResult stat;
         
-        solver.apply(BlockProcessor<GFS,skipBlocksizeCheck>::getVector(z),
-            BlockProcessor<GFS,skipBlocksizeCheck>::getVector(r),stat);
+        solver.apply(z,r,stat);
         stats.tsolve= watch.elapsed();
         res.converged  = stat.converged;
         res.iterations = stat.iterations;
Index: dune/pdelab/backend/seqistlsolverbackend.hh
===================================================================
--- dune/pdelab/backend/seqistlsolverbackend.hh	(revision 2292)
+++ dune/pdelab/backend/seqistlsolverbackend.hh	(working copy)
@@ -543,10 +543,10 @@
         \param[in] r right hand side
         \param[in] reduction to be achieved
       */
-      void apply(M& A, V& z, V& r, typename V::ElementType reduction)
+      void apply(MatrixType& A, V& z, V& r, typename V::ElementType reduction)
       {
         Timer watch;
-        MatrixType& mat=A.base();
+        MatrixType& mat=A;
         typedef Dune::Amg::CoarsenCriterion<Dune::Amg::SymmetricCriterion<MatrixType,
           Dune::Amg::FirstDiagonal> > Criterion;
         SmootherArgs smootherArgs;
