--- orig_tutorialproblem_coupled.hh	2012-09-24 08:56:29.268511049 +0200
+++ ex1d_tutorialproblem_coupled.hh	2012-09-24 10:35:57.793432961 +0200
@@ -42,7 +42,8 @@
 
 // The components that are used
 #include <dumux/material/components/h2o.hh>
-#include <dumux/material/components/lnapl.hh>
+#include <dumux/material/components/brine.hh>
+#include <dumux/material/components/dnapl.hh>
 #include <dumux/common/cubegridcreator.hh>
 
 namespace Dumux{
@@ -71,14 +72,14 @@
 SET_PROP(TutorialProblemCoupled, WettingPhase) /*@\label{tutorial-coupled:2p-system-start}@*/
 {
 private: typedef typename GET_PROP_TYPE(TypeTag, Scalar) Scalar;
-public: typedef Dumux::LiquidPhase<Scalar, Dumux::H2O<Scalar> > type; /*@\label{tutorial-coupled:wettingPhase}@*/
+public: typedef Dumux::LiquidPhase<Scalar, Dumux::Brine<Scalar, Dumux::H2O<Scalar> > >type; /*@\label{tutorial-coupled:wettingPhase}@*/
 };
 
 // Set the non-wetting phase
 SET_PROP(TutorialProblemCoupled, NonwettingPhase)
 {
 private: typedef typename GET_PROP_TYPE(TypeTag, Scalar) Scalar;
-public: typedef Dumux::LiquidPhase<Scalar, Dumux::LNAPL<Scalar> > type; /*@\label{tutorial-coupled:nonwettingPhase}@*/
+public: typedef Dumux::LiquidPhase<Scalar, Dumux::DNAPL<Scalar> > type; /*@\label{tutorial-coupled:nonwettingPhase}@*/
 }; /*@\label{tutorial-coupled:2p-system-end}@*/
 
 SET_TYPE_PROP(TutorialProblemCoupled, FluidSystem, Dumux::TwoPImmiscibleFluidSystem<TypeTag>);/*@\label{tutorial-coupled:set-fluidsystem}@*/
