Index: tutorialspatialparams_coupled.hh
===================================================================
--- tutorialspatialparams_coupled.hh	(revision 15068)
+++ tutorialspatialparams_coupled.hh	(working copy)
@@ -54,10 +54,10 @@
     // material law typedefs
     typedef typename GET_PROP_TYPE(TypeTag, Scalar) Scalar;
     // select material law to be used
-    typedef RegularizedBrooksCorey<Scalar> RawMaterialLaw;     /*@\label{tutorial-coupled:rawlaw}@*/
+//     typedef RegularizedBrooksCorey<Scalar> RawMaterialLaw;     /*@\label{tutorial-coupled:rawlaw}@*/
 public:
     // adapter for absolute law
-    typedef EffToAbsLaw<RawMaterialLaw> type;   /*@\label{tutorial-coupled:eff2abs}@*/
+    typedef LinearMaterial<Scalar> type;   /*@\label{tutorial-coupled:eff2abs}@*/
 };
 }
 
@@ -150,12 +150,12 @@
             K_[i][i] = 1e-7;
 
         //set residual saturations
-        materialParams_.setSwr(0.0);                /*@\label{tutorial-coupled:setLawParams}@*/
-        materialParams_.setSnr(0.0);
+//         materialParams_.setSwr(0.0);                /*@\label{tutorial-coupled:setLawParams}@*/
+//         materialParams_.setSnr(0.0);
 
         //parameters of Brooks & Corey Law
-        materialParams_.setPe(500.0);
-        materialParams_.setLambda(2);
+        materialParams_.setEntryPc(0.0);
+        materialParams_.setMaxPc(2000.0);
     }
 
 private:
