Index: tutorialproblem_coupled.hh
===================================================================
--- tutorialproblem_coupled.hh	(revision 15068)
+++ tutorialproblem_coupled.hh	(working copy)
@@ -127,6 +127,20 @@
         : ParentType(timeManager, gridView)
         , eps_(3e-6)
     {
+        try
+        {
+
+            outflowRateOil_ = GET_RUNTIME_PARAM(TypeTag, Scalar, Problem.OutflowRateOil);
+
+        }
+        catch (Dumux::ParameterException &e) {
+            std::cerr << e << ". Abort!\n";
+            exit(1) ;
+        }
+        catch (...) {
+            std::cerr << "Unknown exception thrown!\n";
+            exit(1);
+        }
 #if !(HAVE_DUNE_ALUGRID || HAVE_UG)
       std::cout << "If you want to use simplices instead of cubes, install and use ALUGrid or UGGrid." << std::endl;
 #endif // !(HAVE_DUNE_ALUGRID || HAVE_UG)
@@ -194,7 +208,7 @@
         if (globalPos[0] > right - eps_) {
             // oil outflux of 30 g/(m * s) on the right boundary.
             values[Indices::contiWEqIdx] = 0;
-            values[Indices::contiNEqIdx] = 3e-2;
+            values[Indices::contiNEqIdx] = outflowRateOil_;
         } else {
             // no-flow on the remaining Neumann-boundaries.
             values[Indices::contiWEqIdx] = 0;
@@ -229,6 +243,7 @@
 private:
     // small epsilon value
     Scalar eps_;
+    Scalar outflowRateOil_;
 };
 }
 
