--- ex1b_tutorialproblem_decoupled.hh	2013-09-30 15:27:31.000000000 +0200
+++ tutorialproblem_decoupled.hh	2013-09-30 15:36:26.961561965 +0200
@@ -41,7 +41,8 @@
 
 // the components that are used
 #include <dumux/material/components/h2o.hh>
-#include <dumux/material/components/lnapl.hh>
+#include <dumux/material/components/dnapl.hh>
+#include <dumux/material/components/brine.hh>
 
 namespace Dumux
 {
@@ -75,7 +76,7 @@
 private:
     typedef typename GET_PROP_TYPE(TypeTag, Scalar) Scalar;
 public:
-    typedef Dumux::LiquidPhase<Scalar, Dumux::H2O<Scalar> > type; /*@\label{tutorial-decoupled:wettingPhase}@*/
+    typedef Dumux::LiquidPhase<Scalar, Dumux::Brine<Scalar, H2O<Scalar> > > type; /*@\label{tutorial-decoupled:wettingPhase}@*/
 };
 
 // Set the non-wetting phase
@@ -84,7 +85,7 @@
 private:
     typedef typename GET_PROP_TYPE(TypeTag, Scalar) Scalar;
 public:
-    typedef Dumux::LiquidPhase<Scalar, Dumux::LNAPL<Scalar> > type; /*@\label{tutorial-decoupled:nonwettingPhase}@*/
+    typedef Dumux::LiquidPhase<Scalar, Dumux::DNAPL<Scalar> > type; /*@\label{tutorial-decoupled:nonwettingPhase}@*/
 }; /*@\label{tutorial-decoupled:2p-system-end}@*/
 
 SET_TYPE_PROP(TutorialProblemDecoupled, EvalCflFluxFunction, Dumux::EvalCflFluxCoats<TypeTag>); /*@\label{tutorial-decoupled:cflflux}@*/
