--- ex1c_tutorialproblem_decoupled.hh	2013-09-30 15:36:26.000000000 +0200
+++ tutorialproblem_decoupled.hh	2013-09-30 15:56:19.162117409 +0200
@@ -43,6 +43,7 @@
 #include <dumux/material/components/h2o.hh>
 #include <dumux/material/components/dnapl.hh>
 #include <dumux/material/components/brine.hh>
+#include <dumux/material/fluidsystems/h2oairfluidsystem.hh>
 
 namespace Dumux
 {
@@ -70,23 +71,24 @@
 //Set the grid creator
 SET_TYPE_PROP(TutorialProblemDecoupled, GridCreator, Dumux::CubeGridCreator<TypeTag>); /*@\label{tutorial-decoupled:set-gridcreator}@*/
 
-// Set the wetting phase
-SET_PROP(TutorialProblemDecoupled, WettingPhase) /*@\label{tutorial-decoupled:2p-system-start}@*/
-{
-private:
-    typedef typename GET_PROP_TYPE(TypeTag, Scalar) Scalar;
-public:
-    typedef Dumux::LiquidPhase<Scalar, Dumux::Brine<Scalar, H2O<Scalar> > > type; /*@\label{tutorial-decoupled:wettingPhase}@*/
-};
-
-// Set the non-wetting phase
-SET_PROP(TutorialProblemDecoupled, NonwettingPhase)
-{
-private:
-    typedef typename GET_PROP_TYPE(TypeTag, Scalar) Scalar;
-public:
-    typedef Dumux::LiquidPhase<Scalar, Dumux::DNAPL<Scalar> > type; /*@\label{tutorial-decoupled:nonwettingPhase}@*/
-}; /*@\label{tutorial-decoupled:2p-system-end}@*/
+SET_TYPE_PROP(TutorialProblemDecoupled, FluidSystem, Dumux::H2OAirFluidSystem<TypeTag>);
+//// Set the wetting phase
+//SET_PROP(TutorialProblemDecoupled, WettingPhase) /*@\label{tutorial-decoupled:2p-system-start}@*/
+//{
+//private:
+//    typedef typename GET_PROP_TYPE(TypeTag, Scalar) Scalar;
+//public:
+//    typedef Dumux::LiquidPhase<Scalar, Dumux::Brine<Scalar, H2O<Scalar> > > type; /*@\label{tutorial-decoupled:wettingPhase}@*/
+//};
+//
+//// Set the non-wetting phase
+//SET_PROP(TutorialProblemDecoupled, NonwettingPhase)
+//{
+//private:
+//    typedef typename GET_PROP_TYPE(TypeTag, Scalar) Scalar;
+//public:
+//    typedef Dumux::LiquidPhase<Scalar, Dumux::DNAPL<Scalar> > type; /*@\label{tutorial-decoupled:nonwettingPhase}@*/
+//}; /*@\label{tutorial-decoupled:2p-system-end}@*/
 
 SET_TYPE_PROP(TutorialProblemDecoupled, EvalCflFluxFunction, Dumux::EvalCflFluxCoats<TypeTag>); /*@\label{tutorial-decoupled:cflflux}@*/
 SET_SCALAR_PROP(TutorialProblemDecoupled, ImpetCFLFactor, 0.95); /*@\label{tutorial-decoupled:cflfactor}@*/
@@ -137,6 +139,7 @@
     {
         //write only every 20th time step to output file
         this->setOutputInterval(20);/*@\label{tutorial-decoupled:outputinterval}@*/
+        GET_PROP_TYPE(TypeTag, FluidSystem)::init();
     }
 
     //! The problem name.
@@ -259,7 +262,7 @@
         values = 0;
         if (globalPos[1] > this->bBoxMax()[1] - eps_)
         {
-            values[nPhaseIdx] = 3e-2;
+            values[nPhaseIdx] = 3e-4;
         }
     }
     //! Initial condition at position globalPos.
