Index: tutorialproblem_decoupled.hh
===================================================================
--- tutorialproblem_decoupled.hh	(revision 15068)
+++ tutorialproblem_decoupled.hh	(working copy)
@@ -135,8 +135,8 @@
     TutorialProblemDecoupled(TimeManager &timeManager, const GridView &gridView)
         : ParentType(timeManager, gridView), eps_(1e-6)/*@\label{tutorial-decoupled:constructor-problem}@*/
     {
-        //write only every 10th time step to output file
-        this->setOutputInterval(10);/*@\label{tutorial-decoupled:outputinterval}@*/
+        //write only every 20th time step to output file
+        this->setOutputInterval(20);/*@\label{tutorial-decoupled:outputinterval}@*/
     }
 
     //! The problem name.
