--- tutorial_decoupled.input	2013-10-11 16:52:41.732964359 +0200
+++ EX5_tutorial_decoupled.input	2013-10-11 16:47:14.000000000 +0200
@@ -9,14 +9,14 @@
 ###############################################################
 
 [TimeManager]
-TEnd = 100000 # duration of the simulation [s]
+TEnd = 1e6 # duration of the simulation [s]
 DtInitial = 10 # initial time step size [s]
 
 [Grid]
-UpperRightX = 300 # x-coordinate of the upper-right corner of the grid [m]
-UpperRightY = 60 # y-coordinate of the upper-right corner of the grid [m]
-NumberOfCellsX = 100 # x-resolution of the grid
-NumberOfCellsY = 1 # y-resolution of the grid
+UpperRightX = 100 # x-coordinate of the upper-right corner of the grid [m]
+UpperRightY = 50 # y-coordinate of the upper-right corner of the grid [m]
+NumberOfCellsX = 20 # x-resolution of the grid
+NumberOfCellsY = 10 # y-resolution of the grid
 
 ###############################################################
 # Simulation restart
