Index: tutorial/tutorialspatialparams_decoupled.hh
===================================================================
--- tutorial/tutorialspatialparams_decoupled.hh	(revision 9210)
+++ tutorial/tutorialspatialparams_decoupled.hh	(working copy)
@@ -104,7 +104,7 @@
      */
     double porosity(const Element& element) const
     {
-        return 0.2;
+        return porosity_;
     }
 
     /*! Return the parameter object for the material law (i.e. Brooks-Corey)
@@ -125,9 +125,22 @@
     TutorialSpatialParamsDecoupled(const GridView& gridView)
     : ParentType(gridView), K_(0)
     {
-        for (int i = 0; i < dim; i++)
-                K_[i][i] = 1e-7;
+         try
+         {
+             for (int i = 0; i < dim; i++)
+                 K_[i][i] = GET_RUNTIME_PARAM(TypeTag, Scalar, SpatialParams.Perm);
 
+             porosity_ = GET_RUNTIME_PARAM(TypeTag, Scalar, SpatialParams.Porosity);
+         }
+         catch (Dumux::ParameterException &e) {
+             std::cerr << e << ". Abort!\n";
+             exit(1) ;
+         }
+         catch (...) {
+             std::cerr << "Unknown exception thrown!\n";
+             exit(1);
+         }
+
         // residual saturations
         materialLawParams_.setSwr(0);
         materialLawParams_.setSnr(0);
@@ -143,6 +156,7 @@
 private:
     MaterialLawParams materialLawParams_;
     FieldMatrix K_;
+    Scalar porosity_;
 };
 
 } // end namespace
