### Eclipse Workspace Patch 1.0
#P workingDumux
Index: tutorial/tutorial_decoupled.input
===================================================================
--- tutorial/tutorial_decoupled.input	(revision 9216)
+++ tutorial/tutorial_decoupled.input	(working copy)
@@ -7,7 +7,14 @@
 ###############################################################
 # Mandatory arguments
 ###############################################################
+[Problem]
+OutflowRateOil = 3e-2 # Rate of oil outflow for the Neumann boundary
+OutputInterval = 5
 
+[SpatialParams]
+Perm = 1e-8 # Peremeability
+Porosity = 0.15
+
 [TimeManager]
 TEnd = 100000 # duration of the simulation [s]
 DtInitial = 10 # initial time step size [s]
@@ -18,6 +25,15 @@
 NumberOfCellsX = 100 # x-resolution of the grid
 NumberOfCellsY = 1 # y-resolution of the grid
 
+##########
+# Tweaks
+##########
+[Impet]
+CFLFactor = 0.8
+
+[Vtk]
+OutputLevel = 3
+
 ###############################################################
 # Simulation restart
 #
