diff --git a/tutorial/tutorial_sequential.input b/tutorial/tutorial_sequential.input
index c435ae2..ca09bb4 100644
--- a/tutorial/tutorial_sequential.input
+++ b/tutorial/tutorial_sequential.input
@@ -1,7 +1,7 @@
 [TimeManager]
-TEnd = 100000 # duration of the simulation [s]
+TEnd = 500000 # duration of the simulation [s]
 DtInitial = 10 # initial time step size [s]
 
 [Grid]
-UpperRight = 300 60 # x-/y-coordinate of the upper-right corner of the grid [m]
-Cells = 100 1 # x-/y-resolution of the grid
+UpperRight = 300 300 # x-/y-coordinate of the upper-right corner of the grid [m]
+Cells = 20 10 # x-/y-resolution of the grid
