Index: tutorialproblem_sequential.hh
===================================================================
--- tutorialproblem_sequential.hh	(revision 15068)
+++ tutorialproblem_sequential.hh	(working copy)
@@ -135,8 +135,8 @@
     TutorialProblemSequential(TimeManager &timeManager, const GridView &gridView)
         : ParentType(timeManager, gridView), eps_(1e-6)/*@\label{tutorial-sequential:constructor-problem}@*/
     {
-        //write only every 10th time step to output file
-        this->setOutputInterval(10);/*@\label{tutorial-sequential:outputinterval}@*/
+        //write only every 20th time step to output file
+        this->setOutputInterval(20);/*@\label{tutorial-sequential:outputinterval}@*/
     }
 
     //! The problem name.
