Index: tutorialproblem_decoupled.hh
===================================================================
--- tutorialproblem_decoupled.hh	(revision 15068)
+++ tutorialproblem_decoupled.hh	(working copy)
@@ -42,7 +42,8 @@
 // the components that are used
 #include <dumux/material/components/h2o.hh>
 #include <dumux/material/components/lnapl.hh>
-
+#include <dumux/material/components/brine.hh>
+#include <dumux/material/fluidsystems/h2oairfluidsystem.hh>
 namespace Dumux
 {
 
@@ -70,24 +71,25 @@
 //Set the grid creator
 SET_TYPE_PROP(TutorialProblemSequential, GridCreator, Dumux::CubeGridCreator<TypeTag>); /*@\label{tutorial-sequential:set-gridcreator}@*/
 
-// Set the wetting phase
-SET_PROP(TutorialProblemSequential, WettingPhase) /*@\label{tutorial-sequential:2p-system-start}@*/
-{
-private:
-    typedef typename GET_PROP_TYPE(TypeTag, Scalar) Scalar;
-public:
-    typedef Dumux::LiquidPhase<Scalar, Dumux::H2O<Scalar> > type; /*@\label{tutorial-sequential:wettingPhase}@*/
-};
+SET_TYPE_PROP(TutorialProblemSequential, FluidSystem, Dumux::H2OAirFluidSystem<TypeTag>);
+// // Set the wetting phase
+// SET_PROP(TutorialProblemSequential, WettingPhase) /*@\label{tutorial-sequential:2p-system-start}@*/
+// {
+// private:
+//     typedef typename GET_PROP_TYPE(TypeTag, Scalar) Scalar;
+// public:
+//     typedef Dumux::LiquidPhase<Scalar, Dumux::H2O<Scalar> > type; /*@\label{tutorial-sequential:wettingPhase}@*/
+// };
+// 
+// // Set the non-wetting phase
+// SET_PROP(TutorialProblemSequential, NonwettingPhase)
+// {
+// private:
+//     typedef typename GET_PROP_TYPE(TypeTag, Scalar) Scalar;
+// public:
+//     typedef Dumux::LiquidPhase<Scalar, Dumux::LNAPL<Scalar> > type; /*@\label{tutorial-sequential:nonwettingPhase}@*/
+// }; /*@\label{tutorial-sequential:2p-system-end}@*/
 
-// Set the non-wetting phase
-SET_PROP(TutorialProblemSequential, NonwettingPhase)
-{
-private:
-    typedef typename GET_PROP_TYPE(TypeTag, Scalar) Scalar;
-public:
-    typedef Dumux::LiquidPhase<Scalar, Dumux::LNAPL<Scalar> > type; /*@\label{tutorial-sequential:nonwettingPhase}@*/
-}; /*@\label{tutorial-sequential:2p-system-end}@*/
-
 SET_TYPE_PROP(TutorialProblemSequential, EvalCflFluxFunction, Dumux::EvalCflFluxCoats<TypeTag>); /*@\label{tutorial-sequential:cflflux}@*/
 SET_SCALAR_PROP(TutorialProblemSequential, ImpetCFLFactor, 0.95); /*@\label{tutorial-sequential:cflfactor}@*/
 
@@ -137,6 +139,7 @@
     {
         //write only every 10th time step to output file
         this->setOutputInterval(10);/*@\label{tutorial-sequential:outputinterval}@*/
+        GET_PROP_TYPE(TypeTag, FluidSystem)::init();
     }
 
     //! The problem name.
@@ -259,7 +262,7 @@
         values = 0;
         if (globalPos[0] > this->bBoxMax()[0] - eps_)
         {
-            values[nPhaseIdx] = 3e-2;
+            values[nPhaseIdx] = 3e-4;
         }
     }
     //! Initial condition at position globalPos.
