Index: tutorialproblem_decoupled.hh
===================================================================
--- tutorialproblem_decoupled.hh	(revision 15171)
+++ tutorialproblem_decoupled.hh	(working copy)
@@ -215,7 +215,7 @@
      */
     void boundaryTypesAtPos(BoundaryTypes &bcTypes, const GlobalPosition& globalPos) const /*@\label{tutorial-sequential:bctype}@*/
     {
-            if (globalPos[0] < this->bBoxMin()[0] + eps_)
+            if (globalPos[1] < this->bBoxMin()[1] + eps_)
             {
                 bcTypes.setDirichlet(pwIdx);
                 bcTypes.setDirichlet(swIdx);
@@ -257,7 +257,7 @@
     void neumannAtPos(PrimaryVariables &values, const GlobalPosition& globalPos) const /*@\label{tutorial-sequential:neumann}@*/
     {
         values = 0;
-        if (globalPos[0] > this->bBoxMax()[0] - eps_)
+        if (globalPos[1] > this->bBoxMax()[1] - eps_)
         {
             values[nPhaseIdx] = 3e-2;
         }
