Index: tutorialspatialparams_sequential.hh
===================================================================
--- tutorialspatialparams_sequential.hh	(revision 15068)
+++ tutorialspatialparams_sequential.hh	(working copy)
@@ -103,7 +103,7 @@
      */
     double porosity(const Element& element) const
     {
-        return 0.2;
+        return porosity_;
     }
 
     /*! Return the parameter object for the material law (i.e. Brooks-Corey)
@@ -124,8 +124,21 @@
     TutorialSpatialParamsSequential(const GridView& gridView)
     : ParentType(gridView), K_(0)
     {
-        for (int i = 0; i < dim; i++)
-                K_[i][i] = 1e-7;
+        try
+         {
+             for (int i = 0; i < dim; i++)
+                 K_[i][i] = GET_RUNTIME_PARAM(TypeTag, Scalar, SpatialParams.Perm);
+ 
+             porosity_ = GET_RUNTIME_PARAM(TypeTag, Scalar, SpatialParams.Porosity);
+         }
+         catch (Dumux::ParameterException &e) {
+             std::cerr << e << ". Abort!\n";
+             exit(1) ;
+         }
+         catch (...) {
+             std::cerr << "Unknown exception thrown!\n";
+             exit(1);
+         }
 
         // residual saturations
         materialLawParams_.setSwr(0);
@@ -142,6 +155,7 @@
 private:
     MaterialLawParams materialLawParams_;
     FieldMatrix K_;
+    Scalar porosity_;
 };
 
 } // end namespace
