Index: tutorialproblem_decoupled.hh
===================================================================
--- tutorialproblem_decoupled.hh	(revision 15068)
+++ tutorialproblem_decoupled.hh	(working copy)
@@ -135,8 +135,23 @@
     TutorialProblemSequential(TimeManager &timeManager, const GridView &gridView)
         : ParentType(timeManager, gridView), eps_(1e-6)/*@\label{tutorial-sequential:constructor-problem}@*/
     {
+         int outputInterval;
+         try
+         {
+
+             outflowRateOil_ = GET_RUNTIME_PARAM(TypeTag, Scalar, Problem.OutflowRateOil);
+             outputInterval = GET_RUNTIME_PARAM(TypeTag, Scalar, Problem.OutputInterval);
+         }
+         catch (Dumux::ParameterException &e) {
+             std::cerr << e << ". Abort!\n";
+             exit(1) ;
+         }
+         catch (...) {
+             std::cerr << "Unknown exception thrown!\n";
+             exit(1);
+         }
         //write only every 10th time step to output file
-        this->setOutputInterval(10);/*@\label{tutorial-sequential:outputinterval}@*/
+        this->setOutputInterval(outputInterval);/*@\label{tutorial-sequential:outputinterval}@*/
     }
 
     //! The problem name.
@@ -259,7 +274,7 @@
         values = 0;
         if (globalPos[0] > this->bBoxMax()[0] - eps_)
         {
-            values[nPhaseIdx] = 3e-2;
+            values[nPhaseIdx] = outflowRateOil_;
         }
     }
     //! Initial condition at position globalPos.
@@ -279,6 +294,7 @@
 
 private:
     const Scalar eps_;
+    Scalar outflowRateOil_;
 };
 } //end namespace
 
