diff --git a/tutorial/tutorialproblem_implicit.hh b/tutorial/tutorialproblem_implicit.hh
index c2d5b13..190efa3 100644
--- a/tutorial/tutorialproblem_implicit.hh
+++ b/tutorial/tutorialproblem_implicit.hh
@@ -52,7 +52,7 @@ SET_PROP(TutorialProblemImplicit, Problem) /*@\label{tutorial-implicit:set-probl
 
 // Set grid and the grid creator to be used
 #if HAVE_DUNE_ALUGRID /*@\label{tutorial-implicit:set-grid}@*/
-SET_TYPE_PROP(TutorialProblemImplicit, Grid, Dune::ALUGrid</*dim=*/2, 2, Dune::cube, Dune::nonconforming>); /*@\label{tutorial-implicit:set-grid-ALU}@*/
+SET_TYPE_PROP(TutorialProblemImplicit, Grid, Dune::ALUGrid</*dim=*/2, 2, Dune::simplex, Dune::nonconforming>); /*@\label{tutorial-implicit:set-grid-ALU}@*/
 #elif HAVE_UG
 SET_TYPE_PROP(TutorialProblemImplicit, Grid, Dune::UGGrid<2>);
 #else
