diff --git a/tutorial/tutorialproblem_implicit.hh b/tutorial/tutorialproblem_implicit.hh
index c2d5b13..ebd73a3 100644
--- a/tutorial/tutorialproblem_implicit.hh
+++ b/tutorial/tutorialproblem_implicit.hh
@@ -35,7 +35,8 @@
 
 // The components that are used
 #include <dumux/material/components/h2o.hh>
-#include <dumux/material/components/lnapl.hh>
+#include <dumux/material/components/brine.hh>
+#include <dumux/material/components/dnapl.hh>
 
 namespace Dumux{
 // Forward declaration of the problem class
@@ -63,14 +64,14 @@ SET_TYPE_PROP(TutorialProblemImplicit, Grid, Dune::YaspGrid<2>);
 SET_PROP(TutorialProblemImplicit, WettingPhase) /*@\label{tutorial-implicit:2p-system-start}@*/
 {
 private: typedef typename GET_PROP_TYPE(TypeTag, Scalar) Scalar;
-public: typedef Dumux::FluidSystems::LiquidPhase<Scalar, Dumux::H2O<Scalar> > type; /*@\label{tutorial-implicit:wettingPhase}@*/
+public: typedef Dumux::FluidSystems::LiquidPhase<Scalar, Dumux::Brine<Scalar, Dumux::H2O<Scalar> > >type; /*@\label{tutorial-implicit:wettingPhase}@*/
 };
 
 // Set the non-wetting phase
 SET_PROP(TutorialProblemImplicit, NonwettingPhase)
 {
 private: typedef typename GET_PROP_TYPE(TypeTag, Scalar) Scalar;
-public: typedef Dumux::FluidSystems::LiquidPhase<Scalar, Dumux::LNAPL<Scalar> > type; /*@\label{tutorial-implicit:nonwettingPhase}@*/
+public: typedef Dumux::FluidSystems::LiquidPhase<Scalar, Dumux::DNAPL<Scalar> > type; /*@\label{tutorial-implicit:nonwettingPhase}@*/
 }; /*@\label{tutorial-implicit:2p-system-end}@*/
 
 SET_TYPE_PROP(TutorialProblemImplicit, FluidSystem, Dumux::TwoPImmiscibleFluidSystem<TypeTag>);/*@\label{tutorial-implicit:set-fluidsystem}@*/
