Index: tutorialproblem_implicit.hh
===================================================================
--- tutorialproblem_implicit.hh	(revision 15068)
+++ tutorialproblem_implicit.hh	(working copy)
@@ -46,7 +46,8 @@
 
 // The components that are used
 #include <dumux/material/components/h2o.hh>
-#include <dumux/material/components/lnapl.hh>
+#include <dumux/material/components/brine.hh>
+#include <dumux/material/components/dnapl.hh>
 #include <dumux/io/cubegridcreator.hh>
 
 namespace Dumux{
@@ -76,7 +77,7 @@
 SET_PROP(TutorialProblemImplicit, WettingPhase) /*@\label{tutorial-implicit:2p-system-start}@*/
 {
 private: typedef typename GET_PROP_TYPE(TypeTag, Scalar) Scalar;
-public: typedef Dumux::FluidSystems::LiquidPhase<Scalar, Dumux::H2O<Scalar> > type; /*@\label{tutorial-implicit:wettingPhase}@*/
+public: typedef Dumux::FluidSystems::LiquidPhase<Scalar, Dumux::Brine<Scalar, Dumux::H2O<Scalar> > >type; /*@\label{tutorial-implicit:wettingPhase}@*/
 };
 
 // Set the non-wetting phase
@@ -83,7 +84,7 @@
 SET_PROP(TutorialProblemImplicit, NonwettingPhase)
 {
 private: typedef typename GET_PROP_TYPE(TypeTag, Scalar) Scalar;
-public: typedef Dumux::FluidSystems::LiquidPhase<Scalar, Dumux::LNAPL<Scalar> > type; /*@\label{tutorial-implicit:nonwettingPhase}@*/
+public: typedef Dumux::FluidSystems::LiquidPhase<Scalar, Dumux::DNAPL<Scalar> > type; /*@\label{tutorial-implicit:nonwettingPhase}@*/
 }; /*@\label{tutorial-implicit:2p-system-end}@*/
 
 SET_TYPE_PROP(TutorialProblemImplicit, FluidSystem, Dumux::TwoPImmiscibleFluidSystem<TypeTag>);/*@\label{tutorial-implicit:set-fluidsystem}@*/
