diff --git a/tutorial/tutorialspatialparams_implicit.hh b/tutorial/tutorialspatialparams_implicit.hh
index 18086e9..541c799 100644
--- a/tutorial/tutorialspatialparams_implicit.hh
+++ b/tutorial/tutorialspatialparams_implicit.hh
@@ -54,7 +54,7 @@ private:
     // material law typedefs
     typedef typename GET_PROP_TYPE(TypeTag, Scalar) Scalar;
     // select material law to be used
-    typedef RegularizedBrooksCorey<Scalar> RawMaterialLaw;     /*@\label{tutorial-implicit:rawlaw}@*/
+    typedef LinearMaterial<Scalar> RawMaterialLaw;     /*@\label{tutorial-implicit:rawlaw}@*/
 public:
     // adapter for absolute law
     typedef EffToAbsLaw<RawMaterialLaw> type;   /*@\label{tutorial-implicit:eff2abs}@*/
@@ -154,8 +154,8 @@ public:
         materialParams_.setSnr(0.0);
 
         //parameters of Brooks & Corey Law
-        materialParams_.setPe(500.0);
-        materialParams_.setLambda(2);
+        materialParams_.setEntryPc(0.0);
+        materialParams_.setMaxPc(2000.0);
     }
 
 private:
